src/behavior.ts

```diff
@@ -430,38 +459,46 @@ export function fromFunction<B>(fn: () => B): Behavior<B> {
 }
 
 /** @private */
-class SwitcherBehavior<A> extends ActiveBehavior<A> {
-  node = new Node(this);
+class SwitcherBehavior<A> extends ActiveBehavior<A>
+  implements BListener, SListener<Behavior<A>> {
+  private bNode = new Node(this);
+  private nNode = new Node(this);
   constructor(
     private b: Behavior<A>,
     next: Future<Behavior<A>> | Stream<Behavior<A>>
   ) {
     super();
-    b.addListener(this.node);
+    this.parents = cons(b);
+    b.addListener(this.bNode);
     this.state = b.state;
     if (this.state === State.Push) {
       this.last = at(b);
     }
-    // FIXME: Using `bind` is hardly optimal for performance.
-    next.subscribe(this.doSwitch.bind(this));
+    // @ts-ignore
+    next.addListener(this.nNode);
```

Comment:
Why ts-ignore?

Reply:
it is not necessary, so i'll remove that